common/utils/src/main/scala/org/apache/spark/util/JsonUtils.scala

```diff
@@ -31,12 +31,21 @@ private[spark] trait JsonUtils {
     .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
 
   def toJsonString(block: JsonGenerator => Unit): String = {
-    val baos = new ByteArrayOutputStream()
-    val generator = mapper.createGenerator(baos, JsonEncoding.UTF8)
-    block(generator)
-    generator.close()
-    baos.close()
-    new String(baos.toByteArray, StandardCharsets.UTF_8)
+    var baos: ByteArrayOutputStream = null
+    var generator: JsonGenerator = null
+    try {
+      baos = new ByteArrayOutputStream()
+      generator = mapper.createGenerator(baos, JsonEncoding.UTF8)
+      block(generator)
+      new String(baos.toByteArray, StandardCharsets.UTF_8)
+    } finally {
+      if (generator != null) {
+        generator.close()
+      }
+      if (baos != null) {
+        baos.close()
```

Comment:
Although this is a good habit, for `ByteArrayOutputStream`, its `close()` method is a no-op.

Reply:
Great reminder, let's use `tryWithResource` to handle it (Only close `generator` - `UTF8JsonGenerator`).
<img width="1057" alt="image" src="https://github.com/user-attachments/assets/1342cb75-60a5-4d17-801e-75ce19ca6dd1">

Thanks @MaxGekk @LuciferYang

Reply:
I would prefer to avoid the assumption about current no-op implementation. How about:
```scala
    tryWithResource(new ByteArrayOutputStream()) { baos =>
      tryWithResource(mapper.createGenerator(baos, JsonEncoding.UTF8)) { generator =>
        block(generator)
        new String(baos.toByteArray, StandardCharsets.UTF_8)
      }
    }
```

Reply:
Updated!

Reply:
Only after `generator` executes `close`, `baos` is the final data, so I have slightly adjusted the above code to:
```scala
    tryWithResource(new ByteArrayOutputStream()) { baos =>
      tryWithResource(mapper.createGenerator(baos, JsonEncoding.UTF8)) { generator =>
        block(generator)
      }
      new String(baos.toByteArray, StandardCharsets.UTF_8)
    }
```